libs/core/kubeagi_core/config.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2024 KubeAGI.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import logging
+import traceback
+
+import yaml
+from kubeagi_core.kube.client import ConfigMap, Secret, SystemDatasource, KubeEnv
+
+logger = logging.getLogger(__name__)
+
+
+class Config:
+    """Operate Kubernetes config"""
+
+    def __init__(self, namespace, name, kubeconfig_path):
+        self._namespace = namespace
+        self._name = name
+        self._kubeconfig_path = kubeconfig_path
+
+    def get_system_datasource(self):
+        """Get the system datasource info in the configmap.
+
+        namespace: namespace;
+        name: config map name
+        """
+        try:
+            configmap_client = ConfigMap(kubeconfig_path=self._kubeconfig_path)
+            config_map = configmap_client.read_namespaced_config_map(
+                namespace=self._namespace, name=self._name
+            )
+
+            config = config_map.data.get("config")
+
+            json_data = yaml.safe_load(config)
```

Comment:
yaml_data???

Reply:
the data read directly is in YAML format. Here, we will convert it to JSON format.